hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -294,6 +295,9 @@ private Option<HoodieRecord> prepareRecord(HoodieRecord<T> hoodieRecord) {
       hoodieRecord.deflate();
       return finalRecordOpt;
     } catch (Exception e) {
+      if (!ignoreWriteFailed) {
+        throw new HoodieInsertException("Error writing record " + hoodieRecord, e);
+      }
```

Comment:
Should we fix it in the handle or when commiting the metadata ?

Reply:
This may not be fix in handle，because would report error when data write error such as schema not consistent

Reply:
> Should we fix it in the handle or when commiting the metadata ?

ok，seem better in commit stage to fix it